hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metrics/prometheus/PrometheusReporter.java

```diff
@@ -18,42 +18,75 @@
 
 package org.apache.hudi.metrics.prometheus;
 
+import org.apache.hudi.common.util.StringUtils;
 import org.apache.hudi.config.HoodieWriteConfig;
 import org.apache.hudi.exception.HoodieException;
 import org.apache.hudi.metrics.MetricsReporter;
 
 import com.codahale.metrics.MetricRegistry;
+import io.prometheus.client.Collector;
 import io.prometheus.client.CollectorRegistry;
 import io.prometheus.client.dropwizard.DropwizardExports;
+import io.prometheus.client.dropwizard.samplebuilder.DefaultSampleBuilder;
+import io.prometheus.client.dropwizard.samplebuilder.SampleBuilder;
 import io.prometheus.client.exporter.HTTPServer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
 
 /**
  * Implementation of Prometheus reporter, which connects to the Http server, and get metrics
  * from that server.
  */
 public class PrometheusReporter extends MetricsReporter {
+  private static final Pattern LABEL_PATTERN = Pattern.compile("\\s*,\\s*");
 
   private static final Logger LOG = LoggerFactory.getLogger(PrometheusReporter.class);
+  private static final Map<Integer, CollectorRegistry> PORT_TO_COLLECTOR_REGISTRY = new HashMap<>();
+  private static final Map<Integer, HTTPServer> PORT_TO_SERVER = new HashMap<>();
```

Comment:
no removal of the port upon reporter instance stop() ?

Reply:
I am stopping all http server instances(after your suggestion). but the entire instance of PrometheusReporter will be invalid or GCed. So, not sure if we need to remove entries from map.

Reply:
hmmm... the stop() is instance level, why stop all httpservers that are owned by other reporters?  or i'm missing sth

Reply:
Only the server corresponding to the port is being stopped in stop(). I have also removed the corresponding entry from PORT_TO_COLLECTOR_REGISTRY.

Reply:
yes that previous comment was for an older version. and it was fixed. and thanks for fixing the collector registry.